### DISCUSSION TITLE
Please respect the calling runtime and PATH.

### DISCUSSION THREAD

**Discussion by Dmole** [npm/feedback#619] [CLI]
Currently npm can not be used if there is more than one node version installed,
because npm will ignore the calling node runtime,
and npm will ignore the node runtime on the PATH,
and npm will refuse to run from node directly,
instead npm will use an arbitrary runtime.

```
sudo which node
/usr/local/bin/node
```

```
sudo /usr/local/bin/node --version
v17.1.0
```

```
sudo /usr/local/bin/node /usr/local/bin/npm install -g node@latest
...
npm ERR! npm WARN npm npm does not support Node.js v10.19.0
...
```

```
node
Welcome to Node.js v17.1.0.
Type ".help" for more information.
> npm install -g node@latest
npm should be run outside of the Node.js REPL, in your normal shell.
(Press Ctrl+D to exit.)
>
```

**Comment by MylesBorins** [comment]
Hey @Dmole how are you install node / npm on your system and what operating system are you using.

In general it is not advised to use sudo for the commands you are using, and it is possible that running sudo is changing the node being used as npm is being run by root (and the defined node for that user).

**Comment by Dmole** [comment]
> how [did] you install node / npm on your system

Something like 
```
apt install -y npm
npm install node@latest
npm install npm@latest
```

>  what operating system are you using

Ubuntu 20.04.3 LTS


> In general it is not advised to use sudo for the commands you are using, and it is possible that running sudo is changing the node being used as npm is being run by root (and the defined node for that user).

per user software update on a multi user system seems silly.
Should the -g flag not have a warning then?

**Reply by ljharb** [reply]
@Dmole npm comes with node; `npm install node` doesn't do what you think it does; `npm install npm` doesn't either, you'd need `npm install -g npm` to update it; and i'd suggest not using apt to install node.

**Comment by Dmole** [comment]
> you'd need npm install -g npm to update it

sudo would be required for --global because  /usr/local or wherever would be root only.
so which is it?

This is what I'm trying to fix;

```
sudo npm outdated -g
Package  Current  Wanted  Latest  Location           Depended by
node      17.1.0  17.2.0  17.2.0  node_modules/node  global
```

**Comment by MylesBorins** [comment]
you shouldn't be using npm to install node. I highly recommend nvm for
managing versions of node locally (including on ubuntu)

https://nvm.sh/

**Comment by Dmole** [comment]
> you shouldn't be using npm to install node.

Why would the node package even exist in npm if is should not be used?

**Reply by ljharb** [reply]
It's fine to use it, but it's not meant to be your main node install. Its use is significantly more obscure than that.

npm comes with node, and isn't installed separately. The first step is to install _node_ - something nvm can help you with.

**Comment by Dmole** [comment]
> The first step is to install node

Looks like I have enough of those installed;
```
which -a node
/home/user/node_modules/.bin/node
/usr/local/bin/node
/usr/bin/node
/bin/node
```

Maybe I should start fresh and see if I don't need node from npm for lighthouse and whatever else I may be using it for.
But if I do need node from npm it should be able to manage its own package properly, and not ignore runtime and PATH (maybe nvm would even result in the same issue).

**Reply by ljharb** [reply]
Nobody needs node _from npm_. Install node *by itself*, which will come with npm.

I would definitely suggest removing every copy of node and npm from your system, and then using something like nvm to install a fresh copy of node (not via apt).

**Reply by Dmole** [reply]
> Nobody needs node _from npm_. 

So the node package should be removed from npm because it's just broken?

**Reply by ljharb** [reply]
No, it’s not broken, it just has a niche use case that isn’t yours.
